### PR TITLE
cleanup: add --old to delete old cache files.

### PR DESCRIPTION
**Pull request by mikemcquaid**
Will be useful on the bots to keep disk space under control. CC @Homebrew/owners for thoughts/review.

### CODE REVIEW

**Comment by xu-cheng** [comment]
Need to regenerate the real manpage , otherwise :+1:

**Comment by chdiza** [comment]
Might I suggest changing the name, perhaps to something like  `--older-than` or `--before`?

Reason: Users think that `cleanup` *already* deletes "old" things, and that's what makes it called "cleanup" to begin with.  ("Old" in the user's mind being "from an old version, or from a package I no longer have installed, etc.").  This is especially true in the case of `cleanup -s`.  What this flag does is to cleanup some-but-not-all "old" files.

The alternate names I suggested have the additional advantage that they unconsciously prime the user to believe that an argument needs to be given with it.

**Comment by DomT4** [comment]
What about stealing from git? ` brew cleanup --prune=now ` and similar.

File: Library/Homebrew/cmd/cleanup.rb [hunk 1]
```diff
@@ -67,6 +67,10 @@ def cleanup_keg keg
   def cleanup_cache
     return unless HOMEBREW_CACHE.directory?
     HOMEBREW_CACHE.children.select(&:file?).each do |file|
+      if (old = ARGV.value "old") && file.mtime < (Time.now - 60*60*24*old.to_i)
```
Comment (jacknagel) [inline Library/Homebrew/cmd/cleanup.rb]: We don't need to re-compute these values for each file.

```rb
old = ARGV.value("old")
time = Time.now - 60 * 60 * 24 * old.to_i

HOMEBREW_CACHE.children.select(&:file?).each do |file|
  if old && file.mtime < time
```

File: Library/Homebrew/cmd/cleanup.rb [hunk 2]
```diff
@@ -67,6 +67,10 @@ def cleanup_keg keg
...
+        cleanup_path(file) { file.unlink }
```
Comment (jacknagel) [inline Library/Homebrew/cmd/cleanup.rb]: `next`?

File: Library/Homebrew/manpages/brew.1.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ Note that these flags should only appear after a command.
   * `cat` <formula>:
     Display the source to <formula>.
 
-  * `cleanup [--force] [-ns]` [<formulae>]:
+  * `cleanup [--force] [--old=X] [-ns]` [<formulae>]:
```
Comment (jacknagel) [inline Library/Homebrew/manpages/brew.1.md]: `X` should be something like `<days>`

File: Library/Homebrew/manpages/brew.1.md [hunk 4]
```diff
@@ -61,6 +61,8 @@ Note that these flags should only appear after a command.
 
     If `--force` is passed, remove out-of-date keg-only brews as well.
 
+    If `--old=X` is remove all cache files older than X days.
```
Comment (jacknagel) [inline Library/Homebrew/manpages/brew.1.md]:
```
If `--old=<days>` is specified, remove cached files older than <days>.
```

File: Library/Homebrew/cmd/cleanup.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/cmd/cleanup.rb]: :+1:

File: Library/Homebrew/cmd/cleanup.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Homebrew/cmd/cleanup.rb]: :+1:

File: Library/Homebrew/manpages/brew.1.md (hunk 3 above)
Comment (mikemcquaid) [inline Library/Homebrew/manpages/brew.1.md]: :+1:

File: Library/Homebrew/manpages/brew.1.md (hunk 4 above)
Comment (mikemcquaid) [inline Library/Homebrew/manpages/brew.1.md]: :+1:

**Comment by mikemcquaid** [comment]
Addressed all comments.

**Pull request closed without merging**